### ISSUE TITLE
FIM Ignore Type

### ISSUE THREAD

**Issue by ozgursuder** [wazuh/wazuh#6745]
Hello there

I am using FIM partition with Wazuh 4.0.

I want it to ignore ~ $ .xlsx files, but with no success

~ $ as ssdd.xlsx

I wonder how can I edit the part below

<ignore type = "sregex">. log $ | .htm $ | .jpg $ | .png $ | .chm $ | .pnf $ | .evtx $ </ignore>

**Comment by ozgursuder** [comment]
There is also a character problem as follows

~ $ ad� cont�bution .xlsx

**Comment by antoniomanuelfr** [comment]
Hi @ozgursuder 

The ignore option can be used in two ways:
- You can ignore an entire path 
- You can use a regex to ignore all paths that match your regex. 
You can find more information in this page of the [documentation](https://documentation.wazuh.com/4.0/user-manual/reference/ossec-conf/syscheck.html#ignore)

To test that everythin is working fine, you can enable the debug mode for syscheck. This can be done by adding `syscheck.debug=2` to the file  `/var/ossec/etc/local_internal_options`  if you are using linux and if you are using windows the path will be `C:\Program Files (x86)\ossec-agent\local_internal_options`

This is the ignore configuration that I'm using: 
```xml
    <!-- File types to ignore -->
    <ignore type="sregex">.log$|.swp$|.xlsx$</ignore>

```
And if I enable the debug mode and check the file `/var/ossec/ossec.log` I can see this message:
```
create_db.c:1214 at fim_check_ignore(): DEBUG: (6205): Ignoring 'file' '/test/test-1.xlsx' due to sregex '.log$|.swp$|.xlsx$'
``` 
And no alert is created.

I hope this information is useful and if you have any doubts, don't hesitate to ask.

Kind regards.

**Comment by ozgursuder** [comment]
I want to import .xlsx files, but I don't want to only import files that come with ~ $ .xlsx at the beginning.

If I do as you mentioned it won't get all xlsx files

**Comment by antoniomanuelfr** [comment]
Add interaction between the database and Windows registries.
Hi @ozgursuder 

That can't be done because the ignore option uses `sregex` syntax and that can't be done (this is the page of the documentation of [sregex](https://documentation.wazuh.com/4.0/user-manual/ruleset/ruleset-xml-syntax/regex.html#sregex-os-match-syntax).

I have been trying to use a custom rule that matches the file name and set its level to 0 (if the level of a rule is 0 the alert is ignored). This also didn't work. 

However, the team has developed support for pcre2 regexes in analysisd rules. This feature has been merged into the master branch and it will be available in the next release. 
I have installed a manager from the master branch and tested it with the following rule:
```xml
  <rule id="100001" level="5">
    <if_group>syscheck</if_group>
    <field name="file" type="pcre2">\$~.*\.xlsx$</field>
    <description>Ignoring file</description>
  </rule>
``` 
And I get this alert: 

```
** Alert 1606729167.28725: - local,syslog,sshd,
2020 Nov 30 09:39:27 (jmv74211-PC) any->syscheck
Rule: 100001 (level 5) -> 'Ignoring file '
File 'c:\test_dir\$~test_file.xlsx' added
Mode: realtime

Attributes:
 - Size: 59
.......
```
If the level of the rule is set to 0, this alert won't appear. 

As I mentioned, this feature is not available in 4.0.

I hope this information is useful.
Kind regards
